vscode/src/testController.ts

```diff
@@ -626,6 +628,50 @@ export class TestController {
     linkedCancellationSource.dispose();
   }
 
+  // When trying to a test file or directory, we need to know which framework is used by tests inside of it to resolve
+  // the command correctly. This method will resolve the first test file with children inside to determine the framework
+  // and then set that to all parents
+  private async discoverFrameworkTag(items: vscode.TestItem[]) {
+    const missingFramework = items.filter((item) => {
+      return !item.tags.some((tag) => tag.id.startsWith("framework"));
+    });
+
+    if (missingFramework.length === 0) {
+      return;
+    }
+
+    for (const item of missingFramework) {
+      let testFileItem = item;
+
+      while (!testFileItem.tags.some((tag) => tag === TEST_FILE_TAG)) {
+        const children: vscode.TestItem[] = [];
+
+        testFileItem.children.forEach((child) => {
+          children.push(child);
+        });
```

Comment:
Wish they added more of the collection APIs on this, like `map`

Reply:
Yeah or at least allowed accessing the first entry.